changelog/15912.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:change
+identity: a request to `/identity/group` that includes `member_group_ids` that contains a cycle will now be responded to with a 400 rather than 500
```

Comment:
Should we be concerned with backwards compatibility?

Reply:
That is mainly why I listed this as a `change`. Our [API docs](https://www.vaultproject.io/api-docs#http-status-codes) mention that a 500 response can be retried. Retrying this request will _always_ fail in its current state. Either the provided `member_group_ids` need to change or a separate request against one of the contained groups will need to be made to remove the potential cycle.

Reply:
Thanks for the explanation. Make sense!